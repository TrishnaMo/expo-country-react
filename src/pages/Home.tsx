import React, { ChangeEvent, useState } from 'react';
import { Container, Box, TextField, Button } from '@material-ui/core';
import { useHistory } from 'react-router-dom';

const Home = () => {
  const [country, setCountry] = useState<string>('');
  const history = useHistory();

  const navigateToCountryDetails = () => {
    history.push(`/country-details/${country}`);
  }

  return (
    <Container
      style={{
        display: 'flex',
        alignItems: 'center',
        height: '100%',
      }}
    >
      <Container maxWidth="xs">
        <Box style={{
          border: '1px solid #ccc',
          padding: 30,
          borderRadius: 6,
        }}>
          <TextField 
            label="Enter Country"
            variant="outlined"
            fullWidth
            onChange={(e:ChangeEvent<HTMLInputElement>)=>{
              setCountry(e.target.value);
            }}
          />
          <Button
            type="submit"
            data-testid="submit"
            disabled={country===''}
            variant="contained"
            color="primary"
            fullWidth
            style={{
              padding: 12,
              marginTop: 30
            }}
            onClick={navigateToCountryDetails}
          >
            Submit
          </Button>
        </Box>
      </Container>
    </Container>
  )
}

export default Home;
