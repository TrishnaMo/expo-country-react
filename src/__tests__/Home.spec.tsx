import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../pages/Home";

describe("Test Home Page", () => {
  it("renders snapshot as expected", () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });
  it("should show a button", () => {
    render(<Home />);
    expect(screen.getByRole("button", { name: "Submit" }))
  });

});
