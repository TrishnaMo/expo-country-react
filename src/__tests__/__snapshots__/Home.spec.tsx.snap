// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page renders snapshot as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
        style="display: flex; align-items: center; height: 100%;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXs"
        >
          <div
            class="MuiBox-root MuiBox-root-1"
            style="border: 1px solid #ccc; padding: 30px; border-radius: 6px;"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Enter Country
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-4"
                  >
                    <span>
                      Enter Country
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
              data-testid="submit"
              disabled=""
              style="padding: 12px; margin-top: 30px;"
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
      style="display: flex; align-items: center; height: 100%;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiBox-root MuiBox-root-1"
          style="border: 1px solid #ccc; padding: 30px; border-radius: 6px;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Enter Country
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4"
                >
                  <span>
                    Enter Country
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
            data-testid="submit"
            disabled=""
            style="padding: 12px; margin-top: 30px;"
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
